std/algorithm/sorting.d

```diff
@@ -994,15 +994,15 @@ if (isRandomAccessRange!Range && !isInfinite!Range &&
         "r and index must be same length for makeIndex.");
     static if (IndexType.sizeof < size_t.sizeof)
     {
-        enforce(r.length <= IndexType.max, "Cannot create an index with " ~
+        enforce(r.length <= size_t(IndexType.max + 1), "Cannot create an index with " ~
             "element type " ~ IndexType.stringof ~ " with length " ~
             to!string(r.length) ~ ".");
     }
 
-    for (IndexType i = 0; i < r.length; ++i)
-    {
-        index[cast(size_t) i] = i;
-    }
+    // Use size_t as loop index to avoid overflow on ++i,
+    // e.g. when squeezing 256 elements into a ubyte index.
+    foreach (i; size_t(0) .. size_t(r.length))
```

Comment:
Note, `r.length` should be `size_t`, so I'm pretty sure `i` would be inferred to be `size_t` without the casts.